lib/providers/memory.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+const LRU = require('lru-cache');
+const micromatch = require('micromatch');
+const yup = require('yup');
+
+const schema = yup.object().shape({
+  ttl: yup.number()
+});
+
+class Memory {
+  constructor(config) {
+    config = schema.validateSync(config, { stripUnknown: true });
+    this.name = 'memory';
+    this.ttl = config.ttl;
+    this.client = new LRU();
+  }
+
+  set(key, value, ttl = this.ttl) {
+    return Promise.resolve(
+      this.client.set(key, value, ttl * 1000 /* convert seconds to milliseconds */)
+    );
+  }
+
+  get(key) {
+    return Promise.resolve(this.client.get(key));
+  }
+
+  has(key) {
+    return Promise.resolve(this.client.has(key));
+  }
+
+  getKeys(pattern = '*') {
+    this.client.prune(); // Delete expired records
+    const keys = this.client.keys().filter(Boolean);
```

Comment:
Why do we need to filter by `Boolean` if we already pruned the expired keys?

Reply:
I ensure that the key is not `undefined` because micromatch will throw an error if there is non-string key.